packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -320,6 +450,33 @@ const extractTemplateLiteral = (node: t.TemplateLiteral, meta: Metadata): CSSOut
 
   css.push({ type: 'unconditional', css: literalResult });
 
+  // Deals with Conditional CSS Rules
+  node.expressions.forEach((prop) => {
+    if (t.isArrowFunctionExpression(prop)) {
+      let resolvedBinding = undefined;
+
+      // Skip this when destructuring in styled interpolation functions
+      // eg border: 5px ${({ borderStyle: bs }) => bs} black;
+      if (!t.isObjectPattern(prop.params[0]) && t.isIdentifier(prop.body)) {
+        resolvedBinding = resolveBindingNode(prop.body.name, meta);
```

Comment:
Should this logic be in `evaluateExpression`?

Reply:
Hey @MonicaOlejniczak it seems that the whole thing can be removed. 
I introduced that check to address an issue with storybook, IIRC the build was failing. One of the previous changes/PRs must have fixed it.